clis/teliod/src/main.rs

```diff
@@ -129,19 +194,169 @@ impl CommandListener {
     }
 }
 
+async fn init_api(config: &mut TeliodDaemonConfig, config_path: String) -> Result<(), TeliodError> {
+    let mut client_config = ClientConfig::new(config).await?;
```

Comment:
I think it's a bit strange to hide registration call inside some function or method - maybe it would be nicer to call it directly here? And you could skip UPDATE on successful registration.